Add financialType_name in param transfer checkout

rubofvil opened the pull request:
This param is used to add the description in some payment processors online and to be set correclty is required in method doPayment/doTransferCheckout.

I saw that is algo set in https://github.com/colemanw/webform_civicrm/blob/7.x-4.x/includes/wf_crm_webform_postprocess.inc#L2046, but in this point is not sending to the processor.

Commit:
Add contributionType_name in param transfer checkout

This param is used to add the description in some payment processors online

mattwire commented:
contributionType_name is a deprecated parameter, if a payment processor requires it then it is the payment processor that should be updated to use financialType

rubofvil commented:
Thanks @mattwire for the comment, i updated the title to make more clear. 
I added both "contributionType_name" and "financialType_name" to make more compatible.

KarinG commented:
Hi Ruben - I believe Matt means to say that this needs to be addressed in the payment processor extension code itself; payment processors typically don’t care if you use your VISA to buy donations, tickets, toques or mittens.

sluc23 commented:
@KarinG our first approach was to include this param in payment processor code ([Redsys](https://github.com/ixiam/com.ixiam.payment.redsys))
But then we saw this line, 
https://github.com/colemanw/webform_civicrm/blob/7.x-4.x/includes/wf_crm_webform_postprocess.inc#L2046

```php
$financialTypeId = $params['financial_type_id'];
$financialTypeDetails = wf_civicrm_api('FinancialType', 'get', array('return' => "is_deductible,name",'id' => $financialTypeId));
// Some payment processors expect this to be set (eg. smartdebit)
$params['financialType_name'] = $financialTypeDetails['values'][$financialTypeId]['name'];
$params['financialTypeID'] = $financialTypeId;
```

which looks like it's been added for one specific payment processor needs. This implementation it's at half-way to send the param to the payment processor, so we thought on completing it.

Please confirm you don't want to include this feature in the module (that in my pov will be useful for all payment processors that need to have the **financial type label** as parameter),  and we'll code it into the Redsys source.

Tx

mattwire commented:
@sluc23 The problem we have in CiviCRM with payment processors is that parameters were never clearly defined and they are different depending on when you submit via event registration, contribution page and webform_civicrm.  So, less parameters is better as it is less work to ensure they remain supported.

The "ideal" way to handle this is to create more `get` functions directly in `CRM_Core_Payment` like this: https://lab.civicrm.org/extensions/stripe/blob/master/CRM/Core/Payment/Stripe.php#L920 (but I haven't submitted that because I didn't write a unit test yet!).

sluc23 commented:
ok, moving this to the payment processor core then. We'll close the PR.
tx for the input

KarinG commented:
@sluc23 - sorry we could not get this in. But it is a better solution you handle this in the payment processor code.
